lms/djangoapps/courseware/context_processor.py

```diff
@@ -0,0 +1,46 @@
+"""
+This is the courseware context_processor module.
+
+This is meant to simplify the process of sending user preferences (espec. time_zone and pref-lang)
+to the templates without having to append every view file.
+
+"""
+from django.core.exceptions import ObjectDoesNotExist
+
+from openedx.core.djangoapps.user_api.errors import UserNotFound
+from openedx.core.djangoapps.user_api.preferences.api import get_user_preferences
+
+RETRIEVABLE_PREFERENCES = {
+    'user_timezone': 'time_zone',
+    'user_language': 'pref-lang'
+}
+
+
+def user_timezone_locale_prefs(request):
+    """
+    Checks if request has an authenticated user.
+    If so, sends set (or none if unset) time_zone and language prefs.
+
+    This interacts with the DateUtils to either display preferred or attempt to determine
+    system/browser set time_zones and languages
+
+    """
+    user_prefs = {
+        'user_timezone': None,
+        'user_language': None,
```

Comment:
Can you set the defaults to empty strings here so you don't need to check for `None` values in all the mako files?